File: src/main/java/com/leet/medium/MaximumXORForEachQuery.go
package main

func getMaximumXor(nums []int, maximumBit int) []int {
  max := (1 << maximumBit) - 1
  size, prev := len(nums), max
  ans := make([]int, size)
  for i := 0; i < size; i++ {
    ans[size - i - 1] = nums[i] ^ prev
    prev = ans[size - i - 1]
  }
  return ans
}

File: src/main/java/com/leet/medium/MaximumXORForEachQuery.java
package com.leet.medium;

public class MaximumXORForEachQuery {
  public int[] getMaximumXor(int[] nums, int maximumBit) {
    int val = (1<<maximumBit) - 1;
    int prev = val;
    int[] ans = new int[nums.length];
    for (int i = 0; i < nums.length; i++) {
      ans[nums.length - i - 1] = prev ^ nums[i];
      prev = ans[nums.length - i - 1];
    }
    return ans;
  }
}
